### PR TITLE
Feature logic search next

### PR DESCRIPTION
**Pull request by nguyenhuy95dn**
Feature logic search next
Link redmine: https://dev.framgia.com/redmine/issues/113283
Task: Logic search

### CODE REVIEW

File: app/src/main/res/layout/fragment_product_search_result.xml [hunk 1]
```diff
@@ -1,19 +1,24 @@
...
     xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:app="http://schemas.android.com/apk/res-auto"
     >
 
     <data>
-
+        <variable
+            name="searchProduct"
+            type="com.framgia.forder.screen.searchproduct.ProductSearchResultFragment"
+            />
```
Comment (daolq3012) [inline app/src/main/res/layout/fragment_product_search_result.xml]: remove this variable

File: app/src/main/res/layout/fragment_product_search_result.xml [hunk 2]
```diff
@@ -1,19 +1,24 @@
...
         <variable
             name="viewModel"
             type="com.framgia.forder.screen.searchproduct.ProductSearchResultViewModel"
             />
-
+        <import type="android.support.v7.widget.LinearLayoutManager"/>
```
Comment (daolq3012) [inline app/src/main/res/layout/fragment_product_search_result.xml]: remove this line

File: app/src/main/java/com/framgia/forder/screen/searchproduct/ProductSearchResultFragment.java [hunk 3]
```diff
@@ -18,6 +20,7 @@
 public class ProductSearchResultFragment extends Fragment {
 
     private ProductSearchResultContract.ViewModel mViewModel;
+    private ProductAdapter mAdapter;
```
Comment (DoanVanToan) [inline app/src/main/java/com/framgia/forder/screen/searchproduct/ProductSearchResultFragment.java]: change to local variable

File: app/src/main/res/layout/fragment_product_search_result.xml [hunk 4]
```diff
@@ -1,19 +1,19 @@
...
 
     <data>
-
         <variable
             name="viewModel"
             type="com.framgia.forder.screen.searchproduct.ProductSearchResultViewModel"
             />
-
     </data>
-
```
Comment (daolq3012) [inline app/src/main/res/layout/fragment_product_search_result.xml]: Please keep this break line

File: app/src/main/res/layout/fragment_product_search_result.xml [hunk 5]
```diff
@@ -1,19 +1,19 @@
...
+    xmlns:app="http://schemas.android.com/apk/res-auto"
     >
 
     <data>
-
         <variable
             name="viewModel"
             type="com.framgia.forder.screen.searchproduct.ProductSearchResultViewModel"
             />
-
```
Comment (daolq3012) [inline app/src/main/res/layout/fragment_product_search_result.xml]: Please keep this break line

File: app/src/main/res/layout/fragment_product_search_result.xml [hunk 6]
```diff
@@ -1,19 +1,19 @@
 <?xml version="1.0" encoding="utf-8"?>
 <layout
     xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:app="http://schemas.android.com/apk/res-auto"
     >
 
     <data>
-
```
Comment (daolq3012) [inline app/src/main/res/layout/fragment_product_search_result.xml]: Please keep this break line

File: app/src/main/java/com/framgia/forder/screen/searchproduct/ProductSearchResultViewModel.java [hunk 7]
```diff
@@ -25,4 +42,27 @@ public void onStop() {
...
+    public void onAddToCart(Product product) {
+        // TODO add to cart
+    }
+
+    @Override
+    public void onItemRecyclerViewClick(Object item) {
+        // TODO view detail product
+    }
+
+    public LayoutManagers.LayoutManagerFactory getLayoutManagers() {
```
Comment (daolq3012) [inline app/src/main/java/com/framgia/forder/screen/searchproduct/ProductSearchResultViewModel.java]: remove this method. It unnecessary

**Pull request merged**